cpp/models/ide_secir/parameters_io.cpp

```diff
@@ -0,0 +1,231 @@
+    }
+
+    // Get (global) support_max to determine how many flows in the past we have to compute.
+    ScalarType global_support_max         = model.get_global_support_max(dt);
+    Eigen::Index global_support_max_index = Eigen::Index(std::ceil(global_support_max / dt));
+
+    // m_transitions should be empty at the beginning.
+    if (model.m_transitions.get_num_time_points() > 0) {
+        model.m_transitions = TimeSeries<ScalarType>((int)InfectionTransition::Count);
+    }
+
+    // The first time we need is -4 * global_support_max.
+    Eigen::Index start_shift = -4 * global_support_max_index;
+    // The last time needed is dependent on the mean stay times in the Exposed compartment and in InfectedNoSymptoms
+    // before transition to InfectedSymptoms.
+    Eigen::Index last_time_index_needed = Eigen::Index(std::ceil(
+        (model.parameters.get<TransitionDistributions>()[Eigen::Index(InfectionTransition::ExposedToInfectedNoSymptoms)]
+             .get_mean(dt) +
+         model.parameters
+             .get<TransitionDistributions>()[Eigen::Index(InfectionTransition::InfectedNoSymptomsToInfectedSymptoms)]
+             .get_mean(dt)) /
+        dt));
+    // Create TimeSeries with zeros.
+    for (Eigen::Index i = start_shift; i <= last_time_index_needed; i++) {
+        // Add time point.
+        model.m_transitions.add_time_point(
+            i * dt, TimeSeries<ScalarType>::Vector::Constant((int)InfectionTransition::Count, 0.));
+    }
+
+    //--- Calculate the flow InfectedNoSymptomsToInfectedSymptoms using the RKI data and store in the m_transitions object.---
+    ScalarType min_offset_needed = std::ceil(
+        model.m_transitions.get_time(0) -
+        1); //Need -1 if first time point is integer and just the floor value if not, therefore use ceil and -1
+    ScalarType max_offset_needed = std::ceil(model.m_transitions.get_last_time());
+
+    bool min_offset_needed_avail = false;
+    bool max_offset_needed_avail = false;
+    // Go through the entries of rki_data and check if date is needed for calculation. Confirmed cases are scaled.
+    // Define dumy variables to store the first and the last index of the TimeSeries where the considered entry of rki_data is potentially needed.
+    Eigen::Index idx_needed_first = 0;
+    Eigen::Index idx_needed_last  = 0;
+    ScalarType time_idx           = 0;
+    for (auto&& entry : rki_data) {
+        int offset = get_offset_in_days(entry.date, date);
+        if (!(offset < min_offset_needed) && !(offset > max_offset_needed)) {
+            if (offset == min_offset_needed) {
+                min_offset_needed_avail = true;
+            }
+            // Smallest index for which the entry is needed is when offset=std::ceil(time_idx).
+            idx_needed_first =
+                Eigen::Index(std::max(std::floor((offset - model.m_transitions.get_time(0) - 1) / dt), 0.));
```

Comment:
Here, comment and definition of `idx_needed_first` do not match. Same for idx_needed_last.